packages/react/src/template.tsx

```diff
@@ -95,7 +95,11 @@ export function BaseTemplate({
       throw new RenderError(`Invalid component: ${stateNode.component}`)
     }
 
-    let componentProps = stateNode.props || {}
+    let componentProps = Object.fromEntries(
+      Object.entries(stateNode.props || {}).map(([propKey, propValue]) => {
+        return [propKey, renderProp(propKey, propValue, stateNode, path)]
+      })
+    )
```

Comment:
We first render props, then we validate them. That way, your validation can check for `onValueChange: function` instead of `onValueChange: ${ event }`, leaking impl. detail into client code. Will cherry-pick as a separate PR as this is good improvement.